utils/voronoi3d.py

```diff
@@ -236,81 +238,271 @@ def calc_bvh_projections(bvh, sites):
+        xmin, xmax, ymin, ymax, zmin, zmax = np.min(aligned_coords[0, :]), np.max(aligned_coords[0, :]), np.min(aligned_coords[1, :]), np.max(aligned_coords[1, :]), np.min(aligned_coords[2, :]), np.max(aligned_coords[2, :])
+
+        rectCoords = lambda x1, y1, z1, x2, y2, z2: np.array([[x1, x1, x2, x2, x1, x1, x2, x2],
+                                                            [y1, y2, y2, y1, y1, y2, y2, y1],
+                                                            [z1, z1, z1, z1, z2, z2, z2, z2]])
+
+        realigned_coords = np.matmul(evec, aligned_coords)
+        realigned_coords += means[:, np.newaxis]
+
+        rrc = np.matmul(evec, rectCoords(xmin, ymin, zmin, xmax, ymax, zmax))
+        rrc += means[:, np.newaxis]
+        rrc = rrc.transpose()
+        return rrc
+
     def get_ridges_per_site(voronoi):
         result = defaultdict(list)
+        sites_list = []
         for ridge_idx in range(len(voronoi.ridge_points)):
             site1_idx, site2_idx = tuple(voronoi.ridge_points[ridge_idx])
+            if (voronoi.points.shape[0]-1)<site1_idx:
+                continue
+            if (voronoi.points.shape[0]-1)<site2_idx:
+                continue
             site1 = voronoi.points[site1_idx]
             site2 = voronoi.points[site2_idx]
             middle = (site1 + site2) * 0.5
             normal = site2 - site1
             plane = PlaneEquation.from_normal_and_point(normal, middle)
+            sites_list.append( (ridge_idx, (site1_idx, site2_idx), (list(site1), list(site2) ), plane, ) )
             result[site1_idx].append(plane)
             result[site2_idx].append(plane)
+        print("sites_list:", len(sites_list))
         return result
 
-    def cut_cell(verts, faces, planes, site, spacing):
-        src_mesh = bmesh_from_pydata(verts, [], faces, normal_update=True)
-        n_cuts = 0
-        for plane in planes:
-            if len(src_mesh.verts) == 0:
-                break
-            geom_in = src_mesh.verts[:] + src_mesh.edges[:] + src_mesh.faces[:]
-
-            plane_co = plane.projection_of_point(site)
-            plane_no = plane.normal.normalized()
-            if plane.side_of_point(site) > 0:
-                plane_no = - plane_no
-
-            plane_co = plane_co - 0.5 * spacing * plane_no
-
-            current_verts = np.array([tuple(v.co) for v in src_mesh.verts])
-            signs = PlaneEquation.from_normal_and_point(plane_no, plane_co).side_of_points(current_verts)
-            #print(f"Plane co {plane_co}, no {plane_no}, signs {signs}")
-            if (signs <= 0).all():# or (signs <= 0).all():
-                continue
+    def get_sites_delaunay_params(delaunay):
+        result = defaultdict(list)
+        ringes = []
+        sites_pair = dict()
+        for simplex in delaunay.simplices:
+            site1_idx, site2_idx, site3_idx, site4_idx = tuple( sorted( [i for i in simplex] ) )
+            ringes+= [tuple( [site1_idx, site2_idx] ),
```

Comment:
1) what is "ringes"? did you mean "rings"? or "ridges"?

2) ringes += itertools.combinations(site_idxs, 2)

Reply:
1. Ridges of Couse! :D (fixed any places with that errata)
2. I never used it! Nice. Fixed. (Commit after all finish)